ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdteststun.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+import logging
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.plugin_utils.eos_validate_state_utils.avdtestbase import AvdTestBase
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AvdTestStun(AvdTestBase):
+    """
+    AvdTestStun class for STUN tests.
+    Validates the configuration of the STUN client, specifically the IPv4 source address and port.
+    """
+
+    anta_module = "anta.tests.stun"
+
+    @cached_property
+    def test_definition(self) -> dict | None:
+        """
+        Generates the proper ANTA test definition for all STUN tests.
+
+        Returns:
+            test_definition (dict): ANTA test definition.
+        """
+        anta_tests = []
+
+        # Check if there are any path groups with STUN configuration
+        if (path_groups := get(self.structured_config, "router_path_selection.path_groups")) is None:
+            LOGGER.info("Path groups are not configured to collect STUN interfaces information. %s is skipped.", self.__class__.__name__)
+            return None
+
+        # Get the interfaces with STUN configuration
+        stun_interfaces = [
+            local_interfaces["name"]
+            for group_idx, path_group in enumerate(path_groups)
+            if self.validate_data(data=path_group, data_path=f"router_path_selection.path_groups.[{group_idx}]", required_keys="local_interfaces")
+            for local_interfaces in path_group["local_interfaces"]
+            if self.validate_data(data=local_interfaces, data_path=f"router_path_selection.path_groups.[{group_idx}]", required_keys="stun.server_profiles")
+        ]
+        if not stun_interfaces:
+            LOGGER.info("No local interface found with STUN configuration. %s is skipped.", self.__class__.__name__)
+            return None
+
+        # Generate the ANTA tests for each interface with STUN configuration
+        for source_interface in stun_interfaces:
+            ip_address = self.get_interface_ip("ethernet_interfaces", source_interface)
+            source_address = ip_address.split("/")[0]
+            source_port = 4500  # TODO: Keeping source port as default 4500. We might need to change it later.
```

Comment:
Let's remove this - there is no way to configure source port in EOS today it seems (I know ANTA test can do it but for now we can't :) )